src/vaults/SectorCrossVault.sol

```diff
@@ -0,0 +1,344 @@
+// SPDX-License-Identifier: AGPL-3.0
+pragma solidity 0.8.16;
+
+import { ERC20 } from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { BatchedWithdraw } from "./ERC4626/BatchedWithdraw.sol";
+import { ERC4626 } from "./ERC4626/ERC4626.sol";
+import { IXAdapter } from "../interfaces/adapters/IXAdapter.sol";
+import { SocketIntegrator } from "../common/SocketIntegrator.sol";
+
+// import "hardhat/console.sol";
+
+contract SectorCrossVault is BatchedWithdraw, SocketIntegrator {
+	enum msgType {
+		NONE,
+		DEPOSIT,
+		REDEEM,
+		REQUESTREDEEM,
+		REQUESTVALUEOFSHARES,
+		EMERGENCYWITHDRAW
+	}
+
+	struct Vault {
+		uint16 chainId;
+		address adapter;
+		bool allowed;
+	}
+
+	struct Request {
+		uint256 timestamp;
+		uint256 chainId;
+		address vault;
+	}
+
+	struct HarvestLedger {
+		uint256 depositValue;
+		bool isOpen;
+		uint256 openIndex;
+		Request[] request;
+	}
+
+	// TODO Implement functions with harvestLock modifier
+
+	// Controls deposits
+	mapping(address => Vault) public depositedVaults;
+
+	// Keep track of depositedVaults
+	// TODO Has to implement push to this array
+	address[] internal vaultsArr;
+
+	// Harvest state
+	HarvestLedger public harvestLedger;
+	// Controls emergency withdraw
+	bool internal emergencyEnabled = false;
+
+	constructor(
+		ERC20 _asset,
+		string memory _name,
+		string memory _symbol,
+		address _owner,
+		address _guardian,
+		address _manager,
+		address _treasury,
+		uint256 _perforamanceFee
+	) ERC4626(_asset, _name, _symbol, _owner, _guardian, _manager, _treasury, _perforamanceFee) {
+		// Not sure if needed
+		// TODO check if needed
+		harvestLedger.openIndex = 0;
+		harvestLedger.isOpen = false;
+		harvestLedger.depositValue = 0;
+	}
+
+	/*/////////////////////////////////////////////////////
+					Cross Vault Interface
+	/////////////////////////////////////////////////////*/
+
+	function depositIntoVaults(address[] calldata vaults, uint256[] calldata amounts)
```

Comment:
it might make things simpler if we combine these two array into a single struct array...
For one we don't have to have `checkInputSize`

Reply:
Agree, done.